### PR TITLE
Fix tests for native torch metrics

### PR DESCRIPTION
**Pull request by cuent**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
It aligns `native_torch_metrics` with `master` and fixes tests for native torch metrics.

### CODE REVIEW

**Comment by Borda** [comment]
it shall be properly rebased on master or do cherry-pick... :rabbit:

**Comment by cuent** [comment]
Please let me know if that is okay 😄

File: tests/metrics/functional/test_classification.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+
+@pytest.mark.parametrize(['pred', 'target', 'expected'], [
+    pytest.param(torch.tensor([0, 1]), torch.tensor([0, 1]), 1.),
+    pytest.param(torch.tensor([1, 0]), torch.tensor([0, 1]), 0.),
+    pytest.param(torch.tensor([1, 1]), torch.tensor([1, 0]), 0.5),
+    pytest.param(torch.tensor([1, 0]), torch.tensor([1, 0]), 1.),
+    pytest.param(torch.tensor([0.5, 0.5]), torch.tensor([1, 0]), 0.5)
+])
+def test_auroc(pred, target, expected):
```
Comment (justusschock) [inline tests/metrics/functional/test_classification.py:271]: can we maybe also add a test for auc separately instead of only testing it indirectly @cuent ?

**Comment by justusschock** [comment]
@cuent can you also change the target branch for this PR from master to native_torch_metrics?
This is not yet ready for master :)

@awaelchi this PR is not supposed to land in master, but on a metric development branch (thus no docstrings so far). It basically fixes the tests.

**Comment by Borda** [comment]
@cuent mind rebase on updated `native_torch_metrics`

**Comment by cuent** [comment]
@Borda As @justusschock said is still a wip for master. It is missing some docs and it seems that some tests as well. So I opened the pull request in way that it will match master:native_torch_metrics. 

@justusschock Yeah I can write the test. :)

**Comment by Borda** [comment]
> @Borda As @justusschock said is still a wip for master. It is missing some docs and it seems that some tests as well. So I opened the pull request in way that it will match master:native_torch_metrics.

so `master` or `native_torch_metrics` there are two different branches :]
I think that I understand, so I rebase this one on `native_torch_metrics`

**Comment by Borda** [comment]
just in case the state before rebase is here - https://github.com/PyTorchLightning/pytorch-lightning/tree/cuent/native_torch_metrics
